### PR TITLE
Suggestion to switch If to Case and renegate function

### PR DESCRIPTION
**Pull request by RussellSpitzer**
Was messing around with this and I was wondering if this was any cleaner. I ended up re-negating in the visitor code like you originally had because the logic was easier in my head

### CODE REVIEW

File: core/src/main/java/org/apache/iceberg/schema/UnionByNameVisitor.java [hunk 1]
```diff
@@ -179,20 +179,15 @@ private void updateColumn(Types.NestedField field, Types.NestedField existingFie
...
-        return false;
-      }
-      if (existingType.typeId() == Type.TypeID.DOUBLE && newType.typeId() == Type.TypeID.FLOAT) {
-        return false;
-      }
-      return true;
+  private boolean compatibleType(Type newType, Type existingType) {
+    switch (existingType.typeId()) {
+      case LONG:
+        return newType.typeId() == Type.TypeID.INTEGER || newType.typeId() == Type.TypeID.LONG;
```
Comment (RussellSpitzer) [inline core/src/main/java/org/apache/iceberg/schema/UnionByNameVisitor.java]: i'm even wondering here if we should have a intrinsic function for the type "isWiderThan(type)" or something ... but that may be overkill for now

Comment (huaxingao) [reply core/src/main/java/org/apache/iceberg/schema/UnionByNameVisitor.java]: can we use [isPromotionAllowed ](https://github.com/apache/iceberg/blob/main/api/src/main/java/org/apache/iceberg/types/TypeUtil.java#L398)

Comment (RussellSpitzer) [reply core/src/main/java/org/apache/iceberg/schema/UnionByNameVisitor.java]: Ah great idea!

Comment (rocco408) [reply core/src/main/java/org/apache/iceberg/schema/UnionByNameVisitor.java]: Indeed excellent idea! 👍

**Review by rocco408 — APPROVED** [review]
This looks cleaner, thanks for the update!

**Pull request merged**